### PR TITLE
Add with_prefix option for getters

### LINKED ISSUE #22: Can we have an option to prefix it with `get_`?
https://github.com/Hoverbear/getset/blob/7b00ca8923a50cc7408f203a1111627ca71de526/src/lib.rs#L61

### PR DESCRIPTION
**Pull request by Hoverbear**
Fix #22.

### CODE REVIEW

File: src/generate.rs [hunk 1]
```diff
@@ -77,12 +77,18 @@ pub fn implement(field: &Field, mode: GenMode, params: GenParams) -> TokenStream
...
-                // `#[get = "pub"]` or `#[set = "pub"]`
+                // `#[get = "pub with_prefix"]` or `#[set = "pub"]`
                 Some(Meta::NameValue(MetaNameValue {
                     lit: Lit::Str(ref s),
                     ..
                 })) => {
-                    let visibility = Ident::new(&s.value(), s.span());
+                    let tokens = s.value();
+                    let visibility =
+                        if let Some(t) = tokens.split(" ").find(|v| v.starts_with("pub")) {
```
Comment (nrc) [inline src/generate.rs]: `crate` is also a valid visibility modifier (and doesn't start with `pub`)

File: src/lib.rs [hunk 2]
```diff
@@ -41,24 +63,55 @@ extern crate proc_macro2;
...
+    attr.interpret_meta().map(|v| v.name())
+}
+
+/// Some users want legacy/compatability.
+/// (Getters are often prefixed with `get_`)
+fn has_with_prefix_set(f: &Field) -> bool {
+    let inner = f
+        .attrs
+        .iter()
+        .filter(|v| attr_name(v).expect("attribute") == "get")
```
Comment (nrc) [inline src/lib.rs]: `"attribute"` is not a very helpful error message

File: src/lib.rs [hunk 3]
```diff
@@ -41,24 +63,55 @@ extern crate proc_macro2;
...
 use generate::{GenMode, GenParams};
 
-#[proc_macro_derive(Getters, attributes(get))]
+fn attr_name(attr: &Attribute) -> Option<Ident> {
+    attr.interpret_meta().map(|v| v.name())
+}
+
+/// Some users want legacy/compatability.
+/// (Getters are often prefixed with `get_`)
+fn has_with_prefix_set(f: &Field) -> bool {
```
Comment (nrc) [inline src/lib.rs]: I think this function name makes sense, but unless you know what is happening it reads a bit like word salad :-). Maybe `has_prefix_attr` or something similar?

File: src/generate.rs (hunk 1 above)
Comment (Hoverbear) [reply src/generate.rs]: Great point!

**Comment by Hoverbear** [comment]
@nrc Reflected. :)

**Pull request merged**